components/calendar/calendar.js

```diff
@@ -419,9 +476,11 @@ class Calendar extends LocalizeStaticMixin(RtlMixin(LitElement)) {
 				const year = day.getFullYear();
 				const month = day.getMonth();
 				const date = day.getDate();
+				const description = `${formatDate(day, {format: 'medium'})}. ${selected ? this.localize('selected') : this.localize('notSelected')}`;
+				// role="gridcell" used for NVDA selected behavior to work properly
+				// d2l-offscreen used instead of aria-label for consistency in announcement across screen readers
 				return html`
 					<td
-						aria-label="${formatDate(day, {format: 'medium'})}"
 						aria-selected="${selected ? 'true' : 'false'}"
```

Comment:
Does `aria-selected` still provide value? Curious why the combination of it and `role="gridcell"` is still needed given that the `grid` role isn't anymore. I see the comment about NVDA... what weird stuff is it doing?

Reply:
Yeah so I've found with NVDA it needs both `aria-selected` and `role="gridcell"` because without both those things it will read "Not Selected" on every cell (even if both `aria-selected` and `role` are not present). I'm guessing this is because they have a tabindex and are "interactive" but I'm not sure.

Reply:
That seems bizarre to me... because without those things isn't it just a normal table? Maybe it's the `role="columnheader"`?

Reply:
Sadly doesn't seem to be. I made a minimal codepen with a simple table and no extra roles, just tabindex, and experimenting with role and aria-selected, and it's still behaving like this.

Reply:
Crazy! Silly NVDA.